rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/action/TbDeviceStateNode.java

```diff
@@ -0,0 +1,104 @@
+import java.util.concurrent.ExecutionException;
+
+@Slf4j
+@RuleNode(
+        type = ComponentType.ACTION,
+        name = "device state",
+        nodeDescription = "Triggers device connectivity events",
+        nodeDetails = "If incoming message originator is a device," +
+                " registers configured event in the Device State Service," +
+                " which sends appropriate message to the Rule Engine. " +
+                "Incoming message is forwarded using the <code>Success</code> chain," +
+                " unless an unexpected error occurs during message processing" +
+                " then incoming message is forwarded using the <code>Failure</code> chain." +
+                "<br>" +
+                "Supported device connectivity events are:" +
+                "<ul>" +
+                "<li>Connect event</li>" +
+                "<li>Disconnect event</li>" +
+                "<li>Activity event</li>" +
+                "<li>Inactivity event</li>" +
+                "</ul>" +
+                "This is particularly useful when device isn't using transports to receive data," +
+                " such as when fetching data from external API or computing new data within the rule chain.",
+        configClazz = TbDeviceStateNodeConfiguration.class,
+        uiResources = {"static/rulenode/rulenode-core-config.js"},
+        configDirective = "tbActionNodeDeviceStateConfig"
+)
+public class TbDeviceStateNode implements TbNode {
+
+    @FunctionalInterface
+    private interface ConnectivityAction {
+
+        void execute(TbContext ctx, TbMsg msg);
+
+    }
+
+    private static final Map<String, ConnectivityAction> EVENT_ACTIONS = Map.of(
+            DataConstants.CONNECT_EVENT,
+            (ctx, msg) -> ctx.getRuleEngineDeviceStateService()
+                    .onDeviceConnect(ctx.getTenantId(), (DeviceId) msg.getOriginator()),
+            DataConstants.ACTIVITY_EVENT,
+            (ctx, msg) -> ctx.getRuleEngineDeviceStateService()
+                    .onDeviceActivity(ctx.getTenantId(), (DeviceId) msg.getOriginator(), System.currentTimeMillis()),
+            DataConstants.INACTIVITY_EVENT,
+            (ctx, msg) -> ctx.getRuleEngineDeviceStateService()
+                    .onDeviceInactivity(ctx.getTenantId(), (DeviceId) msg.getOriginator()),
+            DataConstants.DISCONNECT_EVENT,
+            (ctx, msg) -> ctx.getRuleEngineDeviceStateService()
+                    .onDeviceDisconnect(ctx.getTenantId(), (DeviceId) msg.getOriginator())
+    );
+
+    String event;
+
+    @Override
+    public void init(TbContext ctx, TbNodeConfiguration configuration) throws TbNodeException {
+        event = TbNodeUtils.convert(configuration, TbDeviceStateNodeConfiguration.class).getEvent();
+        if (event == null) {
+            throw new TbNodeException("Event cannot be null!");
+        }
+        if (!EVENT_ACTIONS.containsKey(event)) {
+            throw new TbNodeException("Unsupported event: " + event);
```

Comment:
Use ```throw new TbNodeException("Unsupported event: " + event, true);```. This will mark the error as unrecoverable and rule engine will skip re-init attempts for this node.